pkg/cmd/roachtest/cluster.go

```diff
@@ -3049,3 +3049,41 @@ func (c *clusterImpl) GetPreemptedVMs(
 	}
 	return allPreemptedVMs, nil
 }
+
+// GetHostErrorVMs gets any VMs that were part of the cluster but has a host error.
+func (c *clusterImpl) GetHostErrorVMs(
+	ctx context.Context, l *logger.Logger,
+) ([]vm.PreemptedVM, error) {
+	if c.IsLocal() || !c.spec.UseSpotVMs {
+		return nil, nil
+	}
+
+	cachedCluster, ok := roachprod.CachedCluster(c.name)
```

Comment:
nit: the code from here down in this function looks very similar to `GetPreemptedVMs`, you could extract some of it to a helper function.

Reply:
Did some refactoring